src/js/components/SelectMultiple/SelectionSummary.js

```diff
@@ -68,13 +71,25 @@ const SelectionSummary = ({
     selectedInSearch().length === 0
   );
 
-  const summaryText =
-    value?.length === 0 ||
-    onMore ||
-    !value ||
-    (search !== '' && search !== undefined)
-      ? `${value?.length || 0} selected`
-      : `${value?.length || 0} selected of ${options.length}`;
+  const summaryText = useMemo(() => {
+    if (
+      value?.length === 0 ||
+      onMore ||
+      !value ||
+      (search !== '' && search !== undefined)
+    ) {
+      return format({
+        id: 'selectMultiple.selectedMultipleNonTotal',
+        messages,
+        values: { selectedCount: value?.length },
+      });
+    }
+    return format({
+      id: 'selectMultiple.selectedMultiple',
+      messages,
+      values: { selectedCount: value?.length, totalCount: options.length },
+    });
+  }, [value, onMore, search, format, messages, options.length]);
```

Comment:
Alot of duplication and I'm not sure the useMemo is all that necessary:
```suggestion
  const messageId = (
      value?.length === 0 ||
      onMore ||
      !value ||
      (search !== '' && search !== undefined)
    )
      ? 'selectMultiple.selectedMultiple',
      : 'selectMultiple.selectedMultipleNonTotal';
  const summaryText = format({
      id: messageid,
      messages,
      values: { selectedCount: value?.length, totalCount: options.length },
    });
```

Reply:
Tanks, suggestion was added